advanced/matrix/matrix.py

```diff
@@ -0,0 +1,178 @@
+"""An implementation of the class for the data type 'Matrix'."""
+
+
+class Matrix:
+    """A class used to represent an immutable n*n matrix.
+
+        Attributes:
+            __matrix (tuple): a tuple of tuples representing a matrix.
+    """
+    def __init__(self, matrix_tuples):
+        """Initialize instance of matrix and check for validity."""
+        self.__matrix = matrix_tuples
+        self.is_good_matrix()
+
+    def _check_even(self):
+        """Check that matrix has equal number of rows and columns."""
+        for mat_tuple in self.__matrix:
+            if len(mat_tuple) != len(self.__matrix):
+                raise ValueError("The matrix is not n*n")
+
+    def _check_tuple(self):
+        """Check that matrix is made of a tuple of tuples."""
+        if isinstance(self.__matrix, tuple):
+            if not all(isinstance(mat_tuple, tuple)
+                       for mat_tuple in self.__matrix):
```

Comment:
You should use naming that will be as clear as possible for other people who don't know your code.
For example, instead of mat_tuple, maybe matrix_row (to make it clear what this tuple means)